packages/flutter_tools/test/general.shard/linux/linux_doctor_test.dart

```diff
@@ -2,6 +2,7 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+import 'dart:io' show ProcessException;
```

Comment:
Importing `dart:io` is banned in the tool (I'm pretty sure analysis will catch this). You can import this class via: https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/base/io.dart#L92

Reply:
Interesting, analysis passed. I wonder if we only validate that for non-test code?